### PR TITLE
chore: Added producer and consumer metrics to kafkajs instrumentation

### LINKED ISSUE #2386: Add kafkajs metric `MessageBroker/Kafka/Nodes/<host>:<port>/<Produce|Consume>/Named/<topic>`
### Description
To facilitate linking of entities from Node.js to AWS we have to add new metrics for produce and consumption by host.

### Acceptance Criteria
Every producer send and consumption needs to log a new metric of `MessageBroker/Kafka/Nodes/<host>:<port>/<Produce|Consume>/Named/<topic>`.  For consumption, it should be the host that the message was produced on

### PR DESCRIPTION
**Pull request by jsumners-nr**
This PR resolves #2386.

I needed to refactor things a bit in order to record the brokers on client construction and then utilize those brokers across all of the instrumentation points.

### CODE REVIEW

File: lib/instrumentation/kafkajs/index.js [hunk 1]
```diff
@@ -18,17 +18,20 @@ module.exports = function initialize(agent, kafkajs, _moduleName, shim) {
...
- * @param {Agent} params.agent instance of agent
- * @param {string} params.name name of function getting instrumented
- */
-function recordMethodMetric({ agent, name }) {
-  agent.metrics.getOrCreateMetric(`${KAFKA.PREFIX}/${name}`).incrementCallCount()
+  shim.wrap(kafkajs, 'Kafka', function nrConstructorWrapper(shim, orig) {
+    return function nrConstructor() {
+      const params = shim.argsToArray.apply(shim, arguments)
+      // eslint-disable-next-line new-cap
+      const instance = new orig(...params)
```
Comment (bizob2828) [inline lib/instrumentation/kafkajs/index.js:26]: if you still want to wrap I'd recommend you use `Reflect.construct(orig, arguments)`. The other option is to do `shim.wrapClass`

```js
  shim.wrapClass(kafkajs, 'Kafka', new ClassWrapSpec({
    post: function nrConstructorWrapper(shim, wrappedClass, name, args) {
      this[kafkaCtx] = {
        brokers: args[0].brokers
      }

      shim.wrap(this, 'producer', instrumentProducer)
      shim.wrap(this, 'consumer', instrumentConsumer)

    }
  }))
```

File: lib/instrumentation/kafkajs/index.js [hunk 2]
```diff
@@ -18,17 +18,20 @@ module.exports = function initialize(agent, kafkajs, _moduleName, shim) {
...
-/**
- * Convenience method for logging the tracking metrics for producer and consumer
- *
- * @param {object} params to function
- * @param {Agent} params.agent instance of agent
- * @param {string} params.name name of function getting instrumented
- */
-function recordMethodMetric({ agent, name }) {
-  agent.metrics.getOrCreateMetric(`${KAFKA.PREFIX}/${name}`).incrementCallCount()
+  shim.wrap(kafkajs, 'Kafka', function nrConstructorWrapper(shim, orig) {
```
Comment (bizob2828) [inline lib/instrumentation/kafkajs/index.js:22]: good idea abstracting this

File: test/versioned/kafkajs/kafka.tap.js [hunk 3]
```diff
@@ -50,69 +50,78 @@ tap.afterEach(async (t) => {
   await t.context.producer.disconnect()
 })
 
-tap.test('send records correctly', (t) => {
-  t.plan(7)
+try {
```
Comment (bizob2828) [inline test/versioned/kafkajs/kafka.tap.js:53]: not sure i understand this try/catch. was this meant to be removed after you found your issue?

**Review by bizob2828 — CHANGES_REQUESTED** [review]
overall looks good. a suggestion on wrapping the class and a question about a test try/catch

File: test/versioned/kafkajs/kafka.tap.js (hunk 3 above)
Comment (jsumners-nr) [reply test/versioned/kafkajs/kafka.tap.js:53]: Yep. Debug leftover.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**